src/module.ts

```diff
@@ -0,0 +1,39 @@
+import { DynamicModule, Module, Provider } from '@nestjs/common';
+import { ConfigClass, ConfigOptions } from './options';
+import * as core from './core';
+
+@Module({})
+export class ConfigModule {
+  static forRoot(options: ConfigOptions): DynamicModule {
```

Comment:
don't forget to add optional dependency `Logger` for logging errors of validation/transformation